### PR TITLE
CODENVY-542: use Provider for inject  ProjectRegistry in MavenWorkspace

### PR DESCRIPTION
**Pull request by vparfonov**
Signed-off-by: Vitaly Parfonov vparfonov@codenvy.com
@skabashnyuk @riuvshin

### CODE REVIEW

File: plugins/plugin-maven/che-plugin-maven-server/src/test/java/org/eclipse/che/plugin/maven/server/BaseTest.java [hunk 1]
```diff
@@ -57,6 +60,8 @@
 import java.util.Set;
 
 import static org.eclipse.che.plugin.maven.shared.MavenAttributes.MAVEN_ID;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
```
Comment (benoitf) [inline plugins/plugin-maven/che-plugin-maven-server/src/test/java/org/eclipse/che/plugin/maven/server/BaseTest.java]: Why do we have all these new imports ?

Comment (vparfonov) [inline plugins/plugin-maven/che-plugin-maven-server/src/test/java/org/eclipse/che/plugin/maven/server/BaseTest.java]: fixed, thx

**Comment by vparfonov** [comment]
@evidolob ?

**Comment by riuvshin** [comment]
ci-build

**Comment by evidolob** [comment]
OK

**Pull request merged**